print-someip.c

```diff
@@ -0,0 +1,134 @@
+#include "config.h"
+#endif
+
+#include "netdissect-stdinc.h"
+#include "netdissect.h"
+#include "extract.h"
+#include "udp.h"
+
+/*
+ * SOMEIP Header (R19-11)
+ *
+ *     0                   1                   2                   3
+ *     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *    |               Message ID (Service ID/Method ID)               |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *    |                           Length                              |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ 
+ *    |               Request ID (Client ID/Session ID)               |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ 
+ *    | Protocol Ver  | Interface Ver | Message Type  |  Return Code  |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ 
+ *    |                            Payload                            |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ 
+ */
+
+struct tok message_type_values[] = {
+    { 0x00, "REQUEST" },
+    { 0x01, "REQUEST_NO_RETURN" },
+    { 0x02, "NOTIFICATION" },
+    { 0x80, "RESPONSE" },
+    { 0x81, "ERROR" },
+    { 0x20, "TP_REQUEST" },
+    { 0x21, "TP_REQUEST_NO_RETURN" },
+    { 0x22, "TP_NOTIFICATION" },
+    { 0xa0, "TP_RESPONSE" },
+    { 0xa1, "TP_ERROR" },
+    { 0, NULL }
+};
+
+struct tok return_code_values[] = {
+    { 0x00, "E_OK" },
+    { 0x01, "E_NOT_OK" },
+    { 0x02, "E_UNKNOWN_SERVICE" },
+    { 0x03, "E_UNKNOWN_METHOD" },
+    { 0x04, "E_NOT_READY" },
+    { 0x05, "E_NOT_REACHABLE" },
+    { 0x06, "E_TIMEOUT" },
+    { 0x07, "E_WRONG_PROTOCOL_VERSION" },
+    { 0x08, "E_WRONG_INTERFACE_VERSION" },
+    { 0x09, "E_MALFORMED_MESSAGE" },
+    { 0x0a, "E_WRONG_MESSAGE_TYPE" },
+    { 0x0b, "E_E2E_REPEATED" },
+    { 0x0c, "E_E2E_WRONG_SEQUENCE" },
+    { 0x0d, "E_E2E" },
+    { 0x0e, "E_E2E_NOT_AVAILABLE" },
+    { 0x0f, "E_E2E_NO_NEW_DATA" },
+};
+
+void
+someip_print(netdissect_options *ndo, const u_char *bp, u_int len)
+{
+    uint32_t message_id;
+    uint16_t service_id;
+    uint16_t method_or_event_id;
+    uint8_t event_flag;
+    uint32_t message_len;
+    uint32_t request_id;
+    uint16_t client_id;
+    uint16_t session_id;
+    uint8_t protocol_version;
+    uint8_t interface_version;
+    uint8_t message_type;
+    uint8_t return_code;
+
+    ndo->ndo_protocol = "someip";
+
+    if (len < 16) {
+        nd_print_trunc(ndo);
+        return;
+    }
+
+    message_id = GET_BE_U_4(bp);
+    service_id = message_id >> 16;
+    event_flag = (message_id & 0x00008000) >> 15;
+    method_or_event_id = message_id & 0x00007FFF;
+
+    bp += 4;
+
+    message_len = GET_BE_U_4(bp);
+    bp += 4;
+
+    request_id = GET_BE_U_4(bp);
+    client_id = request_id >> 16;
+    session_id = request_id & 0x0000FFFF;
+    bp += 4;
+
+    protocol_version = *bp;
```

Comment:
Please use `GET_U_1()` when fetching single-byte unsigned integer values.  That way, it does bounds checking on the packet data.

Reply:
Done in 246ca11. Thanks. Ciao